### ISSUE TITLE
Indentation check regression with "new"

### ISSUE THREAD

**Issue by mjpt777** [checkstyle/checkstyle#9719]
```java
package checkstyle;

public class Test
{
    public Integer foo(boolean flag)
    {
        Integer result = flag ?
            new Integer(1) :
            new Integer(2);

        return result;
    }

    public Integer bar(boolean flag)
    {
        return flag ?
            new Integer(1) : new Integer(2);
    }
}
```

```xml
        <module name="Indentation">
            <property name="forceStrictCondition" value="true"/>
        </module>
```

```
[ant:checkstyle] [ERROR] C:\deve\checkstyle\Test.java:8:13: 'new' has incorrect indentation level 12, expected level should be 8. [Indentation]
[ant:checkstyle] [ERROR] C:\deve\checkstyle\Test.java:9:13: 'new' has incorrect indentation level 12, expected level should be 8. [Indentation]
[ant:checkstyle] [ERROR] C:\deve\checkstyle\Test.java:17:13: 'new' has incorrect indentation level 12, expected level should be 8. [Indentation]
```

---

The indentation check works on 8.39 but not with 8.40 or 8.41 of Checkstyle.

---

**Comment by nmancus1** [comment]
@Abhishek-kumar09 is this related to https://github.com/checkstyle/checkstyle/issues/9341 ?

**Comment by Abhishek-kumar09** [comment]
> @Abhishek-kumar09 is this related to #9341 ?

No it was only related to try handler, seems like we have issues with "forceStrictCOndition"=true.
When we want to provide more flexibilty in some checks we always get some errors with true forceStrict.

The issue can be approved since there are no duplicates and it's legit expectation.

**Comment by aryaniiit002** [comment]
there is no issue here 
let me explain 

>Here are three acceptable ways to  format ternary expressions:
```

alpha = (aLongBooleanExpression) ? beta : gamma;  

alpha = (aLongBooleanExpression) ? beta
                                 : gamma;  

alpha = (aLongBooleanExpression)
        ? beta 
        : gamma;
```
and in the given example by @mjpt777 `new` has 4 extra spaces and that's why it's showing error and forceStrictCondition is true, also `lineWrappingIndentation` don't show any error with 

```
{
        return flag ?
        new Integer(1) : new Integer(2);
}
```

**Comment by aryaniiit002** [comment]
what's the issue here ? 

Here are three acceptable ways to  format ternary expressions:
```

alpha = (aLongBooleanExpression) ? beta : gamma;  

alpha = (aLongBooleanExpression) ? beta
                                 : gamma;  

alpha = (aLongBooleanExpression)
        ? beta 
        : gamma;
```
and in the given example by @mjpt777 `new` has 4 extra spaces and that's why it's showing error and forceStrictCondition is true, also `lineWrappingIndentation` don't show any error with 

```
{
    return flag ?
    new Integer(1) : new Integer(2);
}
```

**Comment by mjpt777** [comment]
@aryaniiit002 How are those the only three acceptable ways to format ternary expressions?

The return statement is wrapped in the example and with strict indentation the 4 spaces are not extra.  The ternary expression is wrapped and thus should be indented to show the continuation and that it is not a new expression. This has worked previously and is a regression introduced in 8.40.

**Comment by aryaniiit002** [comment]
yes i think i get it, i am a slow learner, sorry for the trouble

**Comment by SGanguly1999** [comment]
I am on it.

**Comment by rnveach** [comment]
PR was abandoned, anyone is free to take this up.
Last commit was https://github.com/checkstyle/checkstyle/commit/6962a60fe3e1bc559e6ea4bedd47c529e78d1b2c.patch which was very close to being approved.
Remaining discussions at the bottom of https://github.com/checkstyle/checkstyle/pull/9766 .

**Comment by mjpt777** [comment]
Is there anything I can do to help progress this?

**Comment by vyazelenko** [comment]
There is also #5448 related to `forceStrictCondition`.

**Comment by romani** [comment]
@mjpt777 , please send pull request

**Comment by mjpt777** [comment]
> @mjpt777 , please send pull request
There is an outstanding PR with https://github.com/checkstyle/checkstyle/pull/9766

**Comment by Rahulkhinchi03** [comment]
@rnveach sir, I can work on this issue. Kindly clarify what changes you were expecting in:

**Comment by Rahulkhinchi03** [comment]
@romani sir, I can work on this issue. Kindly clarify what changes you were expecting in:
https://github.com/checkstyle/checkstyle/pull/9766/files#diff-0dce12f372cea7c4f2be5a2a899af69b78944a0a59d487170114a3e22b91395eR137-R138

**Comment by romani** [comment]
@rnveach , please answer what you expecting.

**Comment by rnveach** [comment]
If you are continuing from previous PR, discussions of changes I thought should be added were at https://github.com/checkstyle/checkstyle/pull/9766#pullrequestreview-660570823 .
If a new PR is being started from scratch, then I would need to see that to determine.

**Comment by nikosmoum** [comment]
Is there any update on when this will be fixed?

**Comment by nick-mancuso** [comment]
> Is there any update on when this will be fixed?

PRs are welcome, anyone can reuse work from https://github.com/checkstyle/checkstyle/pull/9766, we were close to merging this one.

**Comment by vivek-0509** [comment]
I am on it

**Comment by romani** [comment]
fix is merged
